R/error_checking.R

```diff
@@ -0,0 +1,340 @@
+    }
+
+    # if(class(shape_xy_fields))
+    #   {
+    #
+    # }
+    return(result)
+
+}
+
+
+#' Verify raster data
+#'
+#' This function runs a check on raster inputs and looks for errors
+#' @param raster_data (Raster or Character) - Either the full path string to raster file or a raster object
+#' @param raster_col (Character) - Column name that contains the raster object's output variable
+#' @param raster_band (Numeric) - Future variable for dealing with multi band/time series rasters etc
+#' @param bin_method (Character) - Method or function to use to split continuous data into discrete chunks
+#' @param bins (Numeric) - Number of bins/segments in which to divide the raster
+#' @param convert_zero (Boolean) - Convert raster zero values to NA#
+#' @return (Character) - Returns a success token or an error string if failed
+#' @export
+verify_raster <- function(raster_data , raster_col, raster_band, bin_method, bins, convert_zero)
+{
+    # Verify raster_data and class
+    # looks for  raster_col key field
+    # looks for projection
+    # verify bin method
+    # verify bins
+
+    # Default projection if raster is missing CRS
+    default_projection <- "+proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0"
+
+    result <- "Success"
+
+    # Raster loading - if given a path, use that, else expect an object passed in
+    if(is.null(raster_data))
+    {
+        return("Error: Raster data is NULL")
+    }
+    else if("RasterLayer" %in% class(raster_data))
+    {
+        raster_obj <- raster_data
+    }
+    else if("character" %in% class(raster_data))
+    {
+        if (!file.exists(raster_data))
+        {
+            return("Error: Cannon process raster file: Raster file does not exist or raster path is malformed.")
+        }
+    }
+    else
+    {
+        return("Error: Unrecognized raster_data argument.")
+    }
+
+    # Check raster_col argument to make sure it exists
+    if(is.null(raster_col)) # || raster_obj[raster_col] !exists
+    {
+        return("Error: No raster column defined")
+    }
+
+    # Verify raster CRS and assign default if NA
+    if(is.na(crs(raster_obj)) || is.null(crs(raster_obj)))
+    {
+        raster::crs(raster_obj) <- crs(default_projection)
+        print("Applying default CRS to raster (raster CRS was NA or NULL)")
+    }
+
+    return(result)
+}
+
+
+#' Verify map data
+#'
+#' This function runs a check on map data inputs and looks for errors
+#' @param map_data (Data Frame or Character) - A data frame that contains the output data to map, or alternatively a full path to a CSV
+#' @param data_key_field (Character) - Name of key field in data_obj for merging with shape_data
+#' @param data_col (Character) - Column name that contains the data object's output variable
+#' @return (Character) - Returns a success token or an error string if failed
+#' @export
+verify_data <- function(map_data, data_key_field, data_col)
+{
+    # Future
```

Comment:
add as TODO: in the descriptions

See http://www.doxygen.nl/manual/commands.html#cmdtodo